staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -737,19 +757,24 @@ func (r *Request) request(fn func(*http.Request, *http.Response)) error {
 			r.backoffMgr.UpdateBackoff(r.URL(), err, resp.StatusCode)
 		}
 		if err != nil {
-			// "Connection reset by peer" is usually a transient error.
+			// "Connection reset by peer", "Connection refused" or "apiserver is shutting down" are usually a transient errors.
 			// Thus in case of "GET" operations, we simply retry it.
 			// We are not automatically retrying "write" operations, as
 			// they are not idempotent.
-			if !net.IsConnectionReset(err) || r.verb != "GET" {
+			if r.verb != "GET" {
 				return err
 			}
-			// For the purpose of retry, we set the artificial "retry-after" response.
-			// TODO: Should we clean the original response if it exists?
-			resp = &http.Response{
-				StatusCode: http.StatusInternalServerError,
-				Header:     http.Header{"Retry-After": []string{"1"}},
-				Body:       ioutil.NopCloser(bytes.NewReader([]byte{})),
+			// For connection errors and apiserver shutdown errors retry.
+			if IsRetriableConnectionError(err) {
+				// For the purpose of retry, we set the artificial "retry-after" response.
+				// TODO: Should we clean the original response if it exists?
+				resp = &http.Response{
+					StatusCode: http.StatusInternalServerError,
+					Header:     http.Header{"Retry-After": []string{"1"}},
```

Comment:
1 seems too short

Reply:
1 is pre-existing, and fires on connectionreset errors... 1 second doesn't seem awful for that, but does maybe seem too short for apiserver is shutting down

Reply:
i would guess on apiserver shutdown the client-go retry request and possibly hit different endpoint that is not shutting down (in HA env).

Reply:
Will this retry infinitely? Connection refused could be non-transient as well. We should at least fail eventually.

Reply:
review with ignore whitespace, this is all preexisting.